jsapp/js/components/modalForms/connectProjects.es6

```diff
@@ -0,0 +1,367 @@
+    var data = JSON.stringify({
+      data_sharing: {
+        enabled: !this.state.isShared
+      }
+    });
+
+    if (!this.state.isShared) {
+      let dialog = alertify.dialog('confirm');
+      let opts = {
+        title: `${t('Privacy Notice')}`,
+        message: t('This will attach the full dataset from \"##ASSET_NAME##\" as a background XML file to this form. While not easily visible, it is technically possible for anyone entering data to your form to retrieve and view this dataset. Do not use this feature if \"##ASSET_NAME##\" includes sensitive data.').replaceAll('##ASSET_NAME##', this.props.asset.name),
+        labels: {ok: t('Acknowledge and continue'), cancel: t('Cancel')},
+        onok: (evt, value) => {
+          actions.dataShare.toggleDataSharing(this.props.asset.uid, data);
+          dialog.destroy();
+        },
+        oncancel: () => {
+          dialog.destroy();
+        }
+      };
+      dialog.set(opts).show();
+    } else {
+      actions.dataShare.toggleDataSharing(this.props.asset.uid, data);
+    }
+  }
+
+  /*
+   * Utilities
+   */
+
+  generateAutoname(newParent) {
+    if (newParent) {
+      let autoname = newParent.name;
+      autoname = autoname.toLowerCase().substring(0, 30).replace(/(\ |\.)/g, '_');
+      this.setState({newFilename: autoname});
+    }
+  }
+  generateTruncatedDisplayName(name) {
+    return name.length > 30 ? `${name.substring(0, 30)}...` : name;
+  }
+  generateFilteredAssetList() {
+    let attachedParentUids = [];
+    this.state.attachedParents.forEach((item) => {
+      attachedParentUids.push(item.parentUid)
+    });
+
+    // Filter displayed asset list based on unattached projects
+    return (
+      this.state.sharingEnabledAssets.results.filter(
+        item => !attachedParentUids.includes(item.uid)
+      )
+    );
+  }
+
+ /*
+  * May be useful later for replacing autoname with existing name
+  *
+  * getExteralFilename() {
+  *   let filename = '';
+  *   this.props.asset.content.survey.some((element) => {
+  *     if (element.type === XML_EXTERNAL) {
+  *       filename = element.name;
+  *     }
+  *   });
+  *   return filename;
+  * }
+  */
+
+  /*
+   * Rendering
+   */
+
+  renderLoading(message = t('loading…')) {
+    return (
+      <bem.Loading>
+        <bem.Loading__inner>
+          <i />
+          {message}
+        </bem.Loading__inner>
+      </bem.Loading>
+    );
+  }
+  renderSwitchLabel() {
+    if (this.state.isShared) {
+      return (
+        <ToggleSwitch
+          onChange={this.toggleSharingData.bind(this)}
+          label={t('Data sharing enabled')}
+          checked={this.state.isShared}
+        />
+      );
+    } else {
+      return (
+        <ToggleSwitch
+          onChange={this.toggleSharingData.bind(this)}
+          label={t('Data sharing disabled')}
+          checked={this.state.isShared}
+        />
+      );
+    }
+  }
+  renderSelect(sharingEnabledAssets) {
+    const selectClassNames = ['kobo-select__wrapper'];
```

Comment:
**renderSelect 3/3** 
After moving `sharingEnabledAssets` creation here, make an `if` check for it and if it's empty then return `null` (this will cause the rendering to do nothing, i.e. no html node will be created). On `else` you will have now `            <div className='import-data-form'>` with the `kobo-select__wrapper` inside of it.

Reply:
I think this was a leftover from when I first started putting this together. After moving the creation of `sharingEnabledAssets` to `renderSelect` I realized `this.state.sharingEnabledAssets` will never be `null` by the time it renders this (by then, the request for getting sharing enabled assets will have returned an empty array). This check has been doing nothing :P

I don't think it's bad to render the dropdown and text box even if there are no assets, so instead of improving the guard I'm going to just remove it